File: src/frontend/src/app/components/atendimento/agendamento/agendamento/model/slot-dto.ts
import { Exclude, Expose } from 'class-transformer';

@Exclude()
export class SlotDTO {
  @Expose() dataHoraInicio?: string;
  @Expose() dataHoraFim?: string;
  @Expose() livre?: boolean;
  @Expose() agendamentoId?: string;
  @Expose() atendimentoId?: string;
  @Expose() pacienteNome?: string;
  @Expose() convenioNome?: string;
  @Expose() procedimentoNome?: string;
  @Expose() status?: string;
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/atendimento/agendamento/agendamento/dto/SlotDTO.java
package br.com.grupopipa.gestaointegrada.atendimento.agendamento.agendamento.dto;

import java.time.LocalDateTime;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class SlotDTO {

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm")
    private LocalDateTime dataHoraInicio;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm")
    private LocalDateTime dataHoraFim;

    private boolean livre;

    private UUID agendamentoId;

    private UUID atendimentoId;

    private Long atendimentoNumero;

    private String pacienteNome;

    private String convenioNome;

    private String procedimentoNome;

    private String status;
}
